lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -860,6 +864,31 @@ def _create_deployment_parameters(
                 azure_node_runbook.purchase_plan = self._process_marketplace_image_plan(
                     azure_node_runbook.location, azure_node_runbook.marketplace
                 )
+
+            if azure_node_runbook.marketplace:
+                # by default, set image sriov capability as false
+                # get image sriov capability from API
+                image_sriov_capability = False
+                image_sriov_capability = self._get_image_sriov_capability(
+                    azure_node_runbook.location, azure_node_runbook.marketplace
+                )
+                # when selected size has sriov capability, but image doesn't have sriov
+                # capability, remove sriov feature, and set it as false in node
+                # runbook, then sriov is disabled in template.
+                if (
+                    node.capability.features
+                    and features.Sriov.name() in node.capability.features
+                    and not image_sriov_capability
+                ):
+                    azure_node_runbook.enable_sriov = False
```

Comment:
The information can be put in a local variable, it doesn't need to put back in the runbook.